File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] arr = new String[numbers.length];
        
        for(int i=0; i<numbers.length; i++) 
            arr[i] = String.valueOf(numbers[i]);
        
        Arrays.sort(arr, new Comparator<String>(){
            @Override
            public int compare(String s1, String s2){
                return Integer.parseInt(s2+s1) - Integer.parseInt(s1+s2);
            }});
        
        if(arr[0].equals("0")) return "0";
        
        StringBuilder sb = new StringBuilder("");
        for(String s : arr) sb.append(s);
          
        return sb.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string a, string b){
    if(stoi(a+b)>stoi(b+a)) return true;
    return false;
}
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> tmp;
    for(int n : numbers) tmp.push_back(to_string(n));
    sort(tmp.begin(), tmp.end(), compare);
    for(string s : tmp) answer += s;
    if(answer[0]=='0') return "0";
    return answer;
}